railties/CHANGELOG.md

```diff
@@ -1,3 +1,11 @@
+*   Middlewares are now referenced by name, not class, for move/delete operations.
+
+    Previously if you did `config.middleware.delete(Middleware)` it would not work if the middleware was
+    added to the stack as an instance. Now it will as long as the `self.name` method is defined on the
```

Comment:
Hum, I actually doubt `#name` is often defined. Maybe we should use `klass == m || klass === m` as a matcher? This way you can match a class and all it's instances.

Reply:
Typically items in the stack are classes, eg. `Rails::Rack::Logger`. So it'll be calling `name` on the class, which [is defined](https://ruby-doc.org/core-3.0.1/Module.html#method-i-name).

Reply:
But yes, fair point, this could be a breaking change if someone's made an instance of a new custom middleware and added that to the stack. I think in general that's not a good idea, but do you think I should support it anyway?

Reply:
I think we better be safe than sorry.

Reply:
Okay, I'll do that and remove the changelog since it'll be backward compatible.